lectures/5-Tutorials/SymmetricTensors.md

```diff
@@ -0,0 +1,793 @@
+        a⁺a⁻[s, f] .= sqrt(s.uncoupled[1].charge * f.uncoupled[2].charge)
+    end
+end
+a⁺a⁻
+```
+
+```{code-cell} julia
+:tags: [hide-output]
+a⁻a⁺ = TensorMap(zeros, ComplexF64, V ⊗ V ← V ⊗ V)
+for (s, f) in fusiontrees(a⁻a⁺)
+    if s.uncoupled[1].charge == f.uncoupled[1].charge - 1 && s.uncoupled[2].charge == f.uncoupled[2].charge + 1
+        a⁻a⁺[s, f] .= sqrt(f.uncoupled[1].charge * s.uncoupled[2].charge)
+    end
+end
+a⁻a⁺
+```
+
+```{code-cell} julia
+:tags: [hide-output]
+N = TensorMap(zeros, ComplexF64, V ← V)
+for (s, f) in fusiontrees(N)
+    N[s, f] .= f.coupled.charge
+end
+N
+```
+
+
+### Creation and Annihilation Operators as Symmetric Tensors
+
+Just as in the $\mathbb{Z}_2$ case, it is obvious that we cannot directly construct the creation and annihilation operators as instances of a `TensorMap(..., V ← V)` since they are not invariant under conjugation by the symmetry operator. However, it is possible to construct them as tensor maps using an *auxiliary vector space*, based on the following intuition. The creation operator $a^+$ violates particle number conservation by mapping the occupation number $n$ to $n + 1$. From the point of view of representation theory, this process can be thought af the *fusion* of an `U1Irrep(n)` with an `U1Irrep(1)`, naturally giving the fusion product `U1Irrep(n + 1)`. This means we can represent $a^+$ as a `TensorMap(..., V ← V ⊗ A)`, where the auxiliary vector space `A` is contains the $+1$ irrep with degeneracy 1,  `A = U1Space(1 => 1)`. Similarly, the decrease in occupation number when acting with $a^-$ can be thought of as the *splitting* of an `U1Irrep(n)` into an `U1Irrep(n - 1)` and an `U1Irrep(1)`, leading to a representation in terms of a `TensorMap(..., A ⊗ V ← V)`. Based on these observations, we can represent the matrix elements {eq}`eq:bosonopmatel` as blocks labeled by the $\mathrm{U}(1)$ fusion trees
+
+```{figure} ../_static/SymmetricTensors/bosonops.svg
+:scale: 12%
+:name: bosonops
+```
+
+We can then combine these operators in to get the appropriate Hamiltonian terms,
+
+```{figure} ../_static/SymmetricTensors/bosonham.svg
+:scale: 12%
+:name: bosonham
+```
+
+```{note}
+Although we have made a suggestive distinction between the 'left' and 'right' versions of the operators $a_L^\pm$ and $a_R^\pm$, these can actually be represented by the same tensor map due to the [bosonic braiding style](https://jutho.github.io/TensorKit.jl/latest/lib/sectors/#TensorKit.BraidingStyle) of $\mathrm{U}(1)$ irreps. This is no longer the case when considering fermionic systems where, as a consequence, it is much less clear how to construct symmetric Hamiltonian terms if terms of local symmetric objects.
+```
+
+The explicit construction then looks something like
+
+```{code-cell} julia
+A = U1Space(1 => 1)
+```
+
+```{code-cell} julia
+:tags: [hide-output]
+a⁺ = TensorMap(zeros, ComplexF64, V ← V ⊗ A)
+for (s, f) in fusiontrees(a⁺)
+    a⁺[s, f] .= sqrt(f.coupled.charge)
```

Comment:
I was confused for a second here by the use of `f.coupled`, I would have thought `f.uncoupled[1].charge+1`, or `s.uncoupled[1].charge`. But both are of course equal to the coupled charge.